packages/intrepid/matlab/intrelab/CMakeLists.txt

```diff
@@ -39,159 +55,20 @@ endif()
-      message(FATAL_ERROR "${alib} encountered as TPL with unrecognized file extension.")
-    else() 
-      # remove extension and get library name
-      get_filename_component(libname ${alib} NAME_WLE)
-      string(SUBSTRING ${libname} 3 -1 reduced_libname)
-      set(l_Trilinos_TPL "${l_Trilinos_TPL} -l${reduced_libname}")
-    endif()
-  else()
-    # no extension on library name
-    set(l_Trilinos_TPL "${l_Trilinos_TPL} -l${alib}")
-  endif()
-endforeach()
-
-if(APPLE)
-  if(Trilinos_BUILD_SHARED_LIBS)
-    set(MEX_LD_FLAGS "LDFLAGS=")
-    set(MEX_LD_FLAGS "${MEX_LD_FLAGS} -rpath ${Trilinos_INSTALL_DIR}/lib")
-    set(MEX_LD_FLAGS "${MEX_LD_FLAGS} -rpath ${MATLAB_BIN_DIR}/maci64")
-    set(MEX_LD_FLAGS "${MEX_LD_FLAGS} -shared")
-  endif()
-else()
-  set(MEX_LD_FLAGS "LDFLAGS=")
-  foreach(rp ${CMAKE_INSTALL_RPATH})
-    set(MEX_LD_FLAGS "${MEX_LD_FLAGS} -Wl,-R,${rp}")
-  endforeach()
-  set(MEX_LD_FLAGS "${MEX_LD_FLAGS} -Wl,-R,${MATLAB_BIN_DIR}/glnxa64")
-  set(MEX_LD_FLAGS "${MEX_LD_FLAGS} -Wl,--enable-new-dtags")
-endif()
-
-separate_arguments(ll UNIX_COMMAND ${l_Trilinos})
-set(l_Trilinos ${ll})
-
-separate_arguments(ll UNIX_COMMAND ${l_Trilinos_TPL})
-set(l_Trilinos_TPL ${ll})
-
-if(APPLE)
-  set(MEX_EXTENSION "mexmaci64")
-else()
-  set(MEX_EXTENSION "mexa64")
-endif(APPLE)
-
-# create objs folder
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/objs)
-add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/objs/m2i_helpers.o
-    COMMAND "${MEX}" "CXX=${CMAKE_CXX_COMPILER}" "CXXFLAGS=${MEX_FLAGS}" "-largeArrayDims" -c ${Trilinos_INCLUDE_DIRS} ${Trilinos_TPL_INCLUDE_DIRS} "${CMAKE_CURRENT_SOURCE_DIR}/src/m2i_helpers.cpp"
-    BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/objs/m2i_helpers.o
-    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/src/m2i_helpers.cpp
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/objs
-    )
+set(MOBJS "intrepid_setJacobian;intrepid_setJacobianInv;intrepid_setJacobianDet;intrepid_getNumCubaturePoints;intrepid_getCubature;intrepid_getBasisValues;intrepid_computeCellMeasure;intrepid_HGRADtransformGRAD;intrepid_HGRADtransformVALUE;intrepid_multiplyMeasure;intrepid_integrate;intrepid_mapToPhysicalFrame;intrepid_mapToReferenceFrame;intrepid_scalarMultiplyDataField;intrepid_scalarMultiplyDataData;intrepid_dataIntegral;intrepid_evaluate;intrepid_multiplyVectorDataField;intrepid_dotMultiplyDataField;intrepid_vectorMultiplyDataField;intrepid_dotMultiplyDataData;intrepid_computeEdgeMeasure;intrepid_computeFaceMeasure;intrepid_getCellDeviatoricMat;intrepid_getCellVolumetricMat;intrepid_getPhysicalSideNormals;intrepid_mapToReferenceSubcell;intrepid_tensorMultiplyDataData;intrepid_tensorMultiplyDataField;defGrad;evaluateVectorField;evaluateVectorGradField;externalForceElastostaticsTL;geometricBmat;intrepid_det;intrepid_inverse;intrepid_transpose;materialBmat;strainEulerianLagrangian;stress;tangentStiffnessElastostaticsTL;tensorMultiplyDataField;myMatVecProductDataField;myMatVecProductDataData")
```

Comment:
Note that you can format this list a little better by taking out the quotes and `;` an format to 80 chars or so with:

```
set(MOBJS  intrepid_setJacobian  intrepid_setJacobianInv  intrepid_setJacobianDet
  intrepid_getNumCubaturePoints  intrepid_getCubature  intrepid_getBasisValues
  ...
  myMatVecProductDataField  myMatVecProductDataData
  )
```

That just makes it easier to read without having to scroll off the screen.